backend/src/main/java/fk/prof/backend/aggregator/ProfileWorkInfo.java

```
@@ -98,7 +98,7 @@ private int getSampleCount(Recorder.Wse wse) {
   protected FinalizedProfileWorkInfo buildFinalizedEntity() {
     Map<AggregatedProfileModel.WorkType, Integer> mappedWorkTypeSamples = new HashMap<>();
     for(Map.Entry<Recorder.WorkType, Integer> entry: workTypeSamples.entrySet()) {
-      AggregatedProfileModel.WorkType mappedWorkType = ProtoUtil.mapWorkType(entry.getKey());
+      AggregatedProfileModel.WorkType mappedWorkType = ProtoUtil.mapRecorderToAggregatorWorkType(entry.getKey());
       if(mappedWorkType == null) {
```

Null checks are usually not worth it, because you get more compact code (less code to read and analyze for code-reviews such as this) and equally good diagnostically with NullPtrException.

---

Here, I would actually do a null check, because usually with enum translation I would have relied on the exception thrown by the enum itself. Here proto silently returns null. NullPtrException anyway doesn't tell you what became null in the message.

---

I wanted a more descriptive exception to be thrown rather than a NullPointerException. Presently, the exception will indicate which work type was failed to match, looking at logs makes the mistake clearer to a dev